pilot/cmd/pilot-agent/config.go

```diff
@@ -74,6 +86,18 @@ func constructProxyConfig() (meshconfig.ProxyConfig, error) {
 	return applyAnnotations(proxyConfig, annotations), nil
 }
 
+func determineConcurrencyOption() *types.Int32Value {
+	limit, err := readPodCPULimits()
+	if err == nil && limit > 0 {
+		return &types.Int32Value{Value: int32(math.Ceil(float64(limit) / 1000))}
```

Comment:
What are the units for limit and requests? Just trying to understand the `/1000`.

Reply:
if set `cpu: 1000m` in the pod, it shows up as `1000` in the pod (`divisor: 1m` controls this) which ends up being 1000/1000=1, which means we have 1 cpu, so we set concurrency=1.

Ill add this as a comment

Reply:
added